pkg/kv/kvserver/replica_lifecycle_datadriven_test.go

```diff
@@ -179,34 +179,27 @@ func TestReplicaLifecycleDataDriven(t *testing.T) {
 				}
 				repl := rs.mustGetReplicaDescriptor(t, roachpb.NodeID(1))
 
-				batch := tc.storage.NewBatch()
-				defer batch.Close()
-
-				if initialized {
-					require.NoError(t, kvstorage.WriteInitialRangeState(
-						ctx, batch, batch,
-						rs.desc, repl.ReplicaID, rs.version,
-					))
-				} else {
-					require.NoError(t, kvstorage.CreateUninitializedReplica(
-						ctx, kvstorage.TODOState(batch), batch, 1, /* StoreID */
-						roachpb.FullReplicaID{RangeID: rs.desc.RangeID, ReplicaID: repl.ReplicaID},
-					))
-				}
-				tc.updatePostReplicaCreateState(t, ctx, rs, batch)
+				output := tc.mutate(t, func(batch storage.Batch) {
+					if initialized {
+						require.NoError(t, kvstorage.WriteInitialRangeState(
+							ctx, batch, batch,
+							rs.desc, repl.ReplicaID, rs.version,
+						))
+					} else {
+						require.NoError(t, kvstorage.CreateUninitializedReplica(
+							ctx, kvstorage.TODOState(batch), batch, 1, /* StoreID */
+							roachpb.FullReplicaID{RangeID: rs.desc.RangeID, ReplicaID: repl.ReplicaID},
+						))
+					}
+					tc.updatePostReplicaCreateState(t, ctx, rs, batch)
+				})
 
-				// Print the descriptor and batch output.
 				var sb strings.Builder
-				output, err := print.DecodeWriteBatch(batch.Repr())
-				require.NoError(t, err, "error decoding batch")
 				sb.WriteString(fmt.Sprintf("created replica: %v", repl))
 				if output != "" {
 					sb.WriteString("\n")
 					sb.WriteString(output)
 				}
-				// Commit the batch.
-				err = batch.Commit(true)
-				require.NoError(t, err, "error committing batch")
 				return sb.String()
```

Comment:
Looks like `output` can not be empty here. Squash this whole block into one `return fmt.Sprintf()`?